src/main/webapp/assets/modules/job-listing/jobListingController.js

```diff
@@ -71,9 +86,50 @@ techlooper.controller("jobListingController", function (apiService, $scope, vnwC
   }
 
   $scope.filterJob = function() {
-    var keyword = $scope.searchJob.keyword;
-    var locationId = $scope.searchJob.locationId;
-    var location = vnwConfigService.getLocationText(locationId, "en");
-    $location.path("/job-listing/" + utils.toAscii(keyword) + "+" + locationId + "+" + utils.toAscii(location) + "/1");
+    var keyword = $scope.searchJob.keyword ? utils.toAscii($scope.searchJob.keyword) : "en";
+    var locationId = $scope.searchJob.locationId ? $scope.searchJob.locationId : "";
+    var location = "";
+    if (locationId) {
+      location = vnwConfigService.getLocationText(locationId, "en");
+    }
+    var page = 1;
+
+    //apiService.filterJob(keyword, location, page).success(function(response) {
+    //  $scope.totalPage = response.totalPage;
+    //  $scope.totalJob = response.totalJob;
+    //  $scope.page = response.page;
+    //  $scope.jobs = response.jobs;
+    //});
+
+    var searchPath = $scope.buildSearchPath(keyword, locationId, location, page);
+    $location.path(searchPath);
   }
+
+  $scope.buildSearchPath = function(keyword, locationId, location, page) {
+    var result = "/job-listing/";
+    if (keyword) {
+      result += utils.toAscii(keyword);
+    }
+    if (locationId) {
+      result += "+" + locationId;
+    }
+    if (location) {
+      result += "+" + utils.toAscii(location);
+    }
+    if (page && page > 1) {
+      result += "/" + page;
+    }
+    return result;
+  }
+
+  capitalizeWords : function capitalizeWords(str) {
```

Comment:
'capitalizeWords' was used before it was defined.<br>Label 'capitalizeWords' on function statement.<br>'capitalizeWords' is already defined.